src/IIIFPresentation/API/Features/Manifest/DlcsManifestCoordinator.cs

```diff
@@ -0,0 +1,92 @@
+﻿using API.Features.Storage.Helpers;
+using API.Helpers;
+using API.Infrastructure.Requests;
+using Core;
+using DLCS.API;
+using DLCS.Exceptions;
+using DLCS.Models;
+using Models.API.General;
+using Models.API.Manifest;
+using Models.Database.Collections;
+using Newtonsoft.Json.Linq;
+using Repository;
+
+namespace API.Features.Manifest;
+
+public class DlcsManifestCoordinator(
+    IDlcsApiClient dlcsApiClient,
+    PresentationContext dbContext,
+    ILogger<DlcsManifestCoordinator> logger)
+{
+    /// <summary>
+    /// Carry out any required interactions with DLCS for given <see cref="WriteManifestRequest"/>, this can include
+    /// creating a space and/or creating DLCS batches
+    /// </summary>
+    /// <returns>Tuple of any errors encountered and new Manifest SpaceId</returns>
+    public async Task<(ModifyEntityResult<PresentationManifest, ModifyCollectionType>? error, int? space)> HandleDlcsInteractions(WriteManifestRequest request,
```

Comment:
On one hand I like and use tuples, but this is getting... unwieldy. Perhaps aliasing the type could be used? Not sure if we are on/can jump to C#12, but then you can do `using Point = (int x, int y);` :D

Reply:
Hmm, think I just used refactoring tools and left it as is but agree it's messy. Added `DlcsInteractionResult` to encapsulate result, eased creating and tidied up.